In [3]:
from sklearn.metrics import accuracy_score
import numpy as np
import pandas as pd
import scipy.io
from sklearn.linear_model import LogisticRegression

In [4]:
train = scipy.io.loadmat(r"train.mat")
test = scipy.io.loadmat(r"test.mat")
val = scipy.io.loadmat(r"val.mat")

In [5]:
x_train = train['features']
y_train = train['label'].flatten() 

x_val = val['features']
y_val = val['label'].flatten()

x_test = test['features']
y_test = test['label'].flatten()


In [9]:
clf = LogisticRegression(max_iter=5000, random_state=43)
clf = clf.fit(x_train, y_train)

Problem Faced: Using a small number in max iteration an error occurred>>> 
ConvergenceWarning: lbfgs failed to converge (status=1): STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Found out it's a problem that is solved either by inc the max_iter or changing the solver (trying to make the solver converges)


Validation

In [10]:
y_pred = clf.predict(x_val)
print("Validation accuracy:", accuracy_score(y_val, y_pred))

Validation accuracy: 0.9722222222222222


Testing

In [11]:
y_pred = clf.predict(x_test)
print("Test accuracy:", accuracy_score(y_test, y_pred))

Test accuracy: 0.9694444444444444


Testing accuracy: 0.9694
Required accuracy: 0.95

Logistic Regression model provides the required accuracy